dotnet/src/SemanticKernel/Planning/Plan.cs

```diff
@@ -252,14 +252,20 @@ public async Task<Plan> InvokeNextStepAsync(SKContext context, CancellationToken
 
             // Execute the step
             var functionContext = new SKContext(functionVariables, context.Skills, context.LoggerFactory);
-            var result = await step.InvokeAsync(functionContext, cancellationToken: cancellationToken).ConfigureAwait(false);
-            var resultValue = result.Result.Trim();
 
-            if (result.ErrorOccurred)
+            SKContext result;
+
+            try
             {
-                throw new SKException($"Error occurred while running plan step: {result.LastException?.Message}", result.LastException);
+                result = await step.InvokeAsync(functionContext, cancellationToken: cancellationToken).ConfigureAwait(false);
+            }
+            catch (Exception ex)
+            {
+                throw new SKException($"Error occurred while running plan step: {ex.Message}", ex);
```

Comment:
As discussed, we probably don't need to catch/throw here any longer.

Reply:
The try/catch is removed.